[4.x] Fix SelectColumn crash when using objects as state values

A909M opened the pull request:
<!-- FILL OUT ALL RELEVANT SECTIONS, OR THE PULL REQUEST WILL BE CLOSED. -->

## Description

### Problem:

  SelectColumn crashes with "Cannot access offset of type [ObjectType] on array"  
      error when using object instances as state values, particularly common with:    
      - Spatie Model States
      - Any object used as column state 
      
      
<img width="1017" height="735" alt="Screenshot from 2025-08-23 02-17-42" src="https://github.com/user-attachments/assets/99869a32-71a1-4c73-8c14-9c436c481501" />

<!-- Describe the addressed issue or the need for the new or updated functionality. -->

## Visual changes

<!-- Add screenshots/recordings of before and after. -->

## Functional changes

- [ ] Code style has been fixed by running the `composer cs` command.
- [ ] Changes have been tested to not break existing functionality.
- [ ] Documentation is up-to-date.

A909M committed:
fix(tables): handle object state values in SelectColumn

- Convert object states to string keys to prevent array offset errors
- Preserve BackedEnum handling while adding generic object support

danharrin commented on packages/tables/src/Columns/SelectColumn.php:
```diff
@@ -373,6 +373,8 @@ public function getOptionLabel(bool $withDefault = true): ?string
 
             if ($state instanceof BackedEnum) {
                 $state = $state->value;
+            } elseif (is_object($state)) {
```
Is `$state instanceof Stringable` more appropriate? Does it work for you?

danharrin replied:
Also, is there anything else in the Filament codebase that needs to be updated to support value objects like this? If yes, I am leaning towards not supporting them in general. If no, is there a reason why this is the only place where it is broken?

A909M replied:
> Is `$state instanceof Stringable` more appropriate? Does it work for you?

Yes, `$state instanceof Stringable` is more appropriate and works perfectly. It's more specific. I've updated the code to use this.

A909M replied:
> Also, is there anything else in the Filament codebase that needs to be updated to support value objects like this? If yes, I am leaning towards not supporting them in general. If no, is there a reason why this is the only place where it is broken?

This is the only place that needs this fix. Tables SelectColumn works directly with raw 
  model attributes, so it receives object instances from casts. When `getOptionLabel()` uses the 
  object as an array key (`$options[$state]`), it fails because objects can't be array keys.

A909M committed:
fix(tables): improve object state handling in SelectColumn

- Add Stringable import for type checking
- Change generic object check to target Stringable instances specifically
- Ensures only objects implementing __toString() are converted to string
